spring-cloud-gcp-data-spanner/src/main/java/org/springframework/cloud/gcp/data/spanner/core/SpannerTemplate.java

```diff
@@ -246,29 +268,29 @@ public void upsert(Object object, Optional<Set<String>> includeColumns) {
 
 	@Override
 	public void delete(Object entity) {
-		applyMutationUsingEntity(
-				this.mutationFactory::delete, entity);
+		applyMutationUsingEntity(this.mutationFactory::delete, entity);
+	}
+
+	@Override
+	public void deleteAll(Iterable objects) {
+		this.databaseClient.write(
```

Comment:
Why in this situation you don't use `applyMutationTwoArgs`?
I don't mind it as I don't really like the `applyMutationTwoArgs` design, but just wondering.

Reply:
I think it would look rather the same (in terms of number of lines and key functions called) in this situation because we need to do this extra stream-map call due to mutationFactory::delete returning a single mutation (since we rely on cascading delete) rather than Collection<mutation> (as with the other write operations). 

regarding that other delete function, I think you're right. I'll remove that one. When adding the deleteAll function just now I hadn't noticed we had one that acted similarly already. The new deleteAll is more consistent with the other "all" functions, and I'll keep that one.

Reply:
@meltsufin changes made, PTAL.